src/platform_impl/macos/view.rs

```diff
@@ -884,6 +884,15 @@ impl WinitView {
         WindowId(self.window().id())
     }
 
+    pub(super) fn request_redraw(&self) {
+        if is_main_thread() {
+            // Request a callback via `drawRect`
+            self.setNeedsDisplay(true);
```

Comment:
This has been tried before, if I remember correct it is not enough to make it work in the case where the view is layer backed, see https://github.com/rust-windowing/winit/pull/2189#issuecomment-1035121710. Have you tested it with `wgpu`?

I would suggest fixing the `RedrawRequested` events in another PR, otherwise I suspect this one may drag out for long.

Reply:
Ah, that's awkward. 

Yeah, the general reason for the change was just to do with also wanting to use RedrawRequested as a hard interruption point for `pump_events`, as a mitigation factor for being able to spam input events that might block from returning to the external loop.

That strategy can still be used to cap how long pump_events runs without this change, it's just unfortunate that the RedrawRequested events aren't throttled in any way.

I'll revert this bit so it can be investigated in a separate PR like you say.

Reply:
For metal it looks like you can also manually request redraws explicitly via https://developer.apple.com/documentation/metalkit/mtkview/1535993-enablesetneedsdisplay

I would guess that it could be ok that `setNeedsDisplay` is ignored with a metal/gl view by default, from the pov that you will instead get continuous, regular drawRect callbacks (so request_redraw can be a NOP) but it would be good to investigate this separately - it sounds like it's not that simple.